### ISSUE TITLE
Inquiries about "Rain rendering for evaluating and improving robustness to bad weather"

### ISSUE THREAD

**Issue by wangjiyuan9** [astra-vision/rain-rendering#25]
Dear author,
Hello! I am following your article "Rain rendering for evaluating and improving robustness to bad weather". This is really great work!
I noticed that you mentioned the nuscenes-augment dataset（PBR+GAN） used for downstream task in rainy conditions in your paper. Would it be possible to share it with me? I would like to test my models on this dataset. Thank you!
Also, in the paper, you mentioned using curriculum learning to complete the optimization. What are the details of this part? How was the curriculum set up, and how was the training scheduler set up?
Finally, you stated that "we found that training from scratch with heavy rain or random rainfall rates" fails to converge, but from my observation, both train on heavy rain or on the hybrid method can converge. I don't understand how you arrived at the conclusion in the paper.
Thank you for your time and help. Looking forward to your reply!

**Comment by rdecharette** [comment]
Hello,

thanks for your comment. Sadly, as mentioned we cannot share the PBR+GAN dataset but refer to [this issue](https://github.com/astra-vision/rain-rendering/issues/20) for details on how to reproduce it.

About curriculum learning, we proceeded by training first on original (non-rainy), then finetuning on 25mm/hr, then 50mm/hr, then 100mm/hr. As far as I remember the tests were initially done on semantic segmentation (PSPNet as I recall, with hyperparams described in IJCV Sec. 7.1). If you have attempted a different architectures this could mean our conclusion does not generalize to all. A number of rain research use curriculum but I wouldn't be surprised it trains without that as this is essentially a trick to help convergence.

Best.

**Comment by rdecharette** [comment]
Great.
Extrinsics do not matter for rain rendering, except for camera height as the raindrops are removed when hitting the ground (height=0). Since all KITTI cameras are same height you can ignore that and reuse same simulation for another camera *if you assume the intrinsics are similar*.

Best.
Raoul

**Comment by wangjiyuan9** [comment]
Thanks！
I had began rendered on the right image but found a interesting fact that left and right rainy images' Rain drops are totally same.
like:
image_03:
![image](https://github.com/astra-vision/rain-rendering/assets/47146885/b109590d-9214-44a5-8432-4cf2b807c54c)
image_02:
![image](https://github.com/astra-vision/rain-rendering/assets/47146885/7d82353e-d1a5-4830-bf05-3471022ebab7)
The backgrounds are move a little because the LR vision . But the rain drop are totally same :(
Are there anyway to solve it quickly?(like get the right part rain drops or change some  parameters)
I think maybe I can change [here](https://github.com/astra-vision/rain-rendering/blob/9e994b0cd7e5e210189821345f6655b21c79ca48/common/generator.py#L414-L420) maybe?
Can you give me some suggestions?

**Comment by rdecharette** [comment]
Of course raindrops are located at the same positions because you currently use the same particles. The photometry of raindrop still however differ since its function of input image (as it changes the drops FOV).

If you want to mimic the LR vision, you could indeed shift their image position `values.image_position_start` `values.image_position_end` as a function of their 3D distance to the viewing camera (should be stored in `world_position_start` and `world_position_end`) and the left-right baseline. This should give rather realistic results.

**Comment by wangjiyuan9** [comment]
Sincere thanks for your prompt reply!
However, I'm sorry to say that even though I have tried very hard to understand your response and find the place to modify the code, I still haven't succeeded. Could you please provide some detailed hints?

I think the modification should be made at [here](https://github.com/astra-vision/rain-rendering/blob/9e994b0cd7e5e210189821345f6655b21c79ca48/common/generator.py#L414-L420) to include the rain drops in the image_03 view, but I'm not sure how much to add on the x-coordinate. Additionally, [this function](https://github.com/astra-vision/rain-rendering/blob/9e994b0cd7e5e210189821345f6655b21c79ca48/common/bad_weather.py#L301) should also be modified, right? Finally, what do you mean by "shift their image position values.image_position_start values.image_position_end as a function of their 3D distance to the viewing camera and the "left-right baseline""? How should I modify code to realize it?

If possible, I kindly hope that you could explain the code modification locations as this [answer](https://github.com/astra-vision/rain-rendering/issues/23#issuecomment-1499304077) does.

I am deeply grateful for your help.
